datalakehouse-api/lakehouse_query_api.py

```
@@ -0,0 +1,931 @@
+async def run_in_executor(fn, *args, timeout: float = SPARK_JOB_TIMEOUT):
+    """Run a blocking function in the thread pool with an optional timeout."""
+    loop = asyncio.get_event_loop()
+    future = loop.run_in_executor(_executor, fn, *args)
+    try:
+        return await asyncio.wait_for(future, timeout=timeout)
+    except asyncio.TimeoutError:
+        raise HTTPException(
+            status_code=status.HTTP_504_GATEWAY_TIMEOUT,
+            detail={"status": "error", "code": 504,
+                    "message": f"Spark job exceeded timeout of {timeout}s"}
+        )
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)